web/app/common/constants.ts

```diff
@@ -9,35 +9,43 @@ export enum BuildStatus {
 export type FastlaneStatus =
     'failure'|'success'|'ci_problem'|'pending'|'missing_fastfile';
 
-export function fastlaneStatusToEnum(status: FastlaneStatus): BuildStatus {
-  switch (status) {
-    case 'success':
-      return BuildStatus.SUCCESS;
-    case 'failure':
-      return BuildStatus.FAILED;
-    case 'pending':
-      return BuildStatus.PENDING;
-    case 'missing_fastfile':
-      return BuildStatus.MISSING_FASTFILE;
-    case 'ci_problem':
-      return BuildStatus.INTERNAL_ISSUE;
-    default:
-      throw new Error(`Unknown status type ${status}`);
+export function fastlaneStatusToEnum(status?: FastlaneStatus): BuildStatus {
+  if (status !== null) {
+    switch (status) {
+      case 'success':
+        return BuildStatus.SUCCESS;
+      case 'failure':
+        return BuildStatus.FAILED;
+      case 'pending':
+        return BuildStatus.PENDING;
+      case 'missing_fastfile':
+        return BuildStatus.MISSING_FASTFILE;
+      case 'ci_problem':
+        return BuildStatus.INTERNAL_ISSUE;
+      default:
+        throw new Error(`Unknown status type ${status}`);
+    }
+  } else {
+    return BuildStatus.PENDING; 
   }
 }
 
-export function buildStatusToIcon(status: BuildStatus) {
-  switch (status) {
-    case BuildStatus.SUCCESS:
-      return 'done';
-    case BuildStatus.PENDING:
-      return 'pause_circle_filled';
-    case BuildStatus.FAILED:
-    case BuildStatus.MISSING_FASTFILE:
-      return 'error';
-    case BuildStatus.INTERNAL_ISSUE:
-      return 'warning';
-    default:
-      throw new Error(`Unknown build status ${status}`);
+export function buildStatusToIcon(status?: BuildStatus) {
+  if (status !== null) {
+    switch (status) {
+      case BuildStatus.SUCCESS:
+        return 'done';
+      case BuildStatus.PENDING:
+        return 'pause_circle_filled';
+      case BuildStatus.FAILED:
+      case BuildStatus.MISSING_FASTFILE:
+        return 'error';
+      case BuildStatus.INTERNAL_ISSUE:
+        return 'warning';
+      default:
+        throw new Error(`Unknown build status ${status}`);
+    }
+  } else {
+    return 'pause_circle_filled';
```

Comment:
this is a good icon for now, we might want to differentiate it from the other in the future.

Reply:
Yep, just as a safe fallback case for now 😄